ros_cross_compile/data_collector.py

```diff
@@ -81,34 +83,59 @@ def __init__(self, ros_workspace_dir: Path,
         print('--------------------------------- Collected Data ---------------------------------')
         print('=================================================================================')
         for datum in data_to_print:
             # readable_time = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(datum['timestamp']))
-            readable_time = datetime.utcfromtimestamp(datum['timestamp']).isoformat()
-            if datum['unit'] == Units.Seconds.value:
-                print('{:>12} | {:>35}: {:.2f} {}'.format(readable_time, datum['name'],
-                                                          datum['value'], datum['unit']),
+            readable_time = datetime.utcfromtimestamp(datum.timestamp).isoformat()
+            if datum.unit == Units.Seconds.value:
+                print('{:>12} | {:>35}: {:.2f} {}'.format(readable_time, datum.metric_name,
+                                                          datum.value, datum.unit),
                       end='')
             else:
-                print('{:>12} | {:>35}: {} {}'.format(readable_time, datum['name'],
-                                                      datum['value'], datum['unit']),
+                print('{:>12} | {:>35}: {} {}'.format(readable_time, datum.metric_name,
+                                                      datum.value, datum.unit),
                       end='')
-            if datum['complete']:
+            if datum.complete:
                 print('\n')
             else:
                 print(' {}'.format('incomplete'))
 
-    def write(self, data_collector: DataCollector, print_data: bool):
+    def serialize_to_cloudwatch(self, data: List[Datum], args: Namespace) -> List[Dict]:
+        """
+        Serialize collected datums to fit the AWS Cloudwatch publishing format.
+
+        To get an idea of what the Cloudwatch formatting is like, refer to
+        https://awscli.amazonaws.com/v2/documentation/api/latest/reference/
+        cloudwatch/put-metric-data.html.
+        """
+        def serialize_helper(datum: Dict) -> Dict:
+            datum['MetricName'] = datum.pop('metric_name')
+            datum['Value'] = datum.pop('value')
+            datum['Unit'] = datum.pop('unit')
+            datum['Timestamp'] = datum.pop('timestamp')
+
+            serialized_dimension = [{'Name': 'Complete', 'Value': str(datum.pop('complete'))},
+                                    {'Name': 'Architecture', 'Value': args.arch},
+                                    {'Name': 'OS', 'Value': args.os},
+                                    {'Name': 'ROS Distro', 'Value': args.rosdistro}]
+            datum['Dimensions'] = serialized_dimension
+            return datum
+
+        serialized_data = list(map(lambda d: d._asdict(), data))
+        serialized_data = list(map(serialize_helper, serialized_data))
+        return serialized_data
+
+    def write(self, data_collector: DataCollector, args: Namespace):
```

Comment:
it's not ok to pass around `argparse.Namespace` - it can contain anything so it doesn't necessarily have what you need. require the user to pass the data you need - which is a `Platform` (or `arch`, `os`, `rosdistro` - but `Platform` would be preferable)